pkg/sql/catalog/dbdesc/database_desc.go

```diff
@@ -110,6 +114,11 @@ func (desc *Immutable) GetParentID() descpb.ID {
 	return keys.RootNamespaceID
 }
 
+// IsModified implements the Descriptor interface.
+func (desc *Immutable) IsModified() bool {
```

Comment:
this is a really weird interface, (maybe I'll think of an alternative by the end of the review). It doesn't make any sense for an Immutable descriptor to ever return true to a "IsModified" API

Reply:
I disagree

Imagine 
```sql
CREATE TABLE t(i INT);
BEGIN;
ALTER TABLE t DROP COLUMN I;
SELECT * FROM t; -- this should resolve a modified immutable descriptor 
```

Reply:
I understand that it could happen, but I don't think that the verbiage makes sense.

Reply:
Okay so your gripe isn't the implementation but rather the name and the commenting, yeah? If I went with `IsUncommittedVersion()` would that make it clearer? Also add some commentary about how it works and about the settings in which it is reliable and put those in the interface comment?

Reply:
Yeah. I can't think of a different way to do what you're trying to do.